src/OpenTelemetry/Trace/ActivityProcessor.cs

```diff
@@ -28,13 +28,29 @@ public abstract class ActivityProcessor
         /// Activity start hook.
         /// </summary>
         /// <param name="activity">Instance of activity to process.</param>
-        public abstract void OnStart(Activity activity);
+        public void OnStart(Activity activity)
```

Comment:
Are you open to renaming these? I think the more typical .NET-y style for this would be:
```csharp
public void Start(Activity activity);
protected abstract void OnStart(Activity activity);
```

Reply:
If others are open, I'm open to this. It does mean a changing the public API and would affect anyone out there that has created a custom processor.

Reply:
Both ways are breaking, no?

Today:
```csharp
public abstract void OnStart(Activity activity);
```

First version:
```csharp
public void OnStart(Activity activity);
protected abstract void OnStartInternal(Activity activity);
```

Proposed version:
```csharp
public void Start(Activity activity);
protected abstract void OnStart(Activity activity);
```

Reply:
Duh. 🤦  Yes, you're right.

Reply:
We could move this logic up into ActivityListener to avoid the break. Like what's I'm doing on #969. But I think this is a better long-term design. We don't want a lot of stuff in ActivityListener IMO. I was thinking once this is available, I would re-do how that is done to be in the `Start` or `Stop` implementation of base ActivityProcessor.

Reply:
@CodeBlanch @reyang 

Given that this PR will likely end up not touching the `ActivityProcessor` class there may not be the need for the protected OnStart/OnEnd methods.

**Question: Do we still want to do this name change?** Seems now is a good time since there are already breaking changes going out on the next release.

If we do the name change do we still want:

```csharp
public virtual void Start(Activity activity);
public virtual void End(Activity activity);
```

or what do you think about more closely aligning to the names used by `ActivitySource`

```csharp
public void StartActivity(Activity activity);
public void StopActivity(Activity activity);  // ActivitySource does not actually have this but DiagnosticSource does
```

I'd like to move the name change to a separate PR if we proceed with either of these options.

Reply:
Normally I would struggle with naming things. This time it is easier for me as the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#interface-definition) said we should use `OnStart` and `OnEnd` 😄.

Reply:
Ah ha! Yes I wondered about that right after submitting this question. Great! We won't touch it,

Reply:
@alanwest Separate PR works for me 👍

Reply:
@CodeBlanch no PR necessary 😄. As Reiley pointed out `OnStart` and `OnEnd` are the names specified in the spec for the public API.

Reply:
Got it. Sorry I didn't realize you changed them to Start/OnStart already. That works for me. I think if there wasn't a spec involved something like StartActivity/OnActivityStarted would be better (what you were getting at), but good to stay as close to the spec as possible.